superset/explore/form_data/commands/delete.py

```diff
@@ -47,14 +49,15 @@ def run(self) -> bool:
                 key
             )
             if state:
-                dataset_id = state["dataset_id"]
-                chart_id = state["chart_id"]
-                check_access(dataset_id, chart_id, actor)
+                datasource_id: int = state["datasource_id"]
+                chart_id: Optional[int] = state["chart_id"]
+                datasource_type = DatasourceType(state["datasource_type"])
+                check_access(datasource_id, chart_id, actor, datasource_type)
                 if state["owner"] != get_owner(actor):
                     raise TemporaryCacheAccessDeniedError()
                 tab_id = self._cmd_params.tab_id
                 contextual_key = cache_key(
-                    session.get("_id"), tab_id, dataset_id, chart_id
+                    session.get("_id"), tab_id, datasource_id, chart_id, datasource_type
```

Comment:
todo: I still need to write up something here for deleting old keys. Unless it's ok that it just expires on it's own. The user will get a 404 though.

Reply:
Since we are generating new keys for misses, i think it's fine to let the keys expire. 

I think it would be a lot of work to have another layer of remapping old keys to new keys.

Reply:
Yeah, let's do that, it's only for delete and update. Unless anyone else has thoughts.